staging/src/k8s.io/apiserver/pkg/authentication/token/cache/cached_token_authenticator_test.go

```diff
@@ -272,29 +271,41 @@ func (s *singleBenchmark) run(b *testing.B) {
 }
 
 func (s *singleBenchmark) bench(b *testing.B) {
-	s.b = b
+	// Simulate slowness, qps limit, external service limitation, etc
+	const maxInFlight = 40
+	chokepoint := make(chan struct{}, maxInFlight)
+	// lookup count
+	var lookups uint64
+
 	a := newWithClock(
-		authenticator.TokenFunc(s.lookup),
+		authenticator.TokenFunc(func(ctx context.Context, token string) (*authenticator.Response, bool, error) {
+			atomic.AddUint64(&lookups, 1)
```

Comment:
While I do not think it matters, this is technically `defer func() { atomic.AddUint64(&lookups, 1) }()` right since we want to count a lookup after it is complete?

Reply:
Actually no that is wrong, you just want to count that you attempted to do a lookup at all.

Reply:
Yep. I'm trying to gauge "how many expensive things do we do per op" so that we can compare the goodness (TM) of our cache strategy.